plugins/flatpak/gs-flatpak.c

```diff
@@ -2499,10 +2499,23 @@ gs_flatpak_get_list_for_remove (GsFlatpak *self, GsApp *app,
 	for (guint i = 0; i < related->len; i++) {
 		FlatpakRelatedRef *xref_related = g_ptr_array_index (related, i);
 		g_autoptr(GsApp) app_tmp = NULL;
+		g_autoptr(GError) error_local = NULL;
+
 		if (!flatpak_related_ref_should_delete (xref_related))
 			continue;
 		app_tmp = gs_flatpak_create_app (self, FLATPAK_REF (xref_related));
 		gs_app_set_origin (app_tmp, gs_app_get_origin (app));
+		if (!gs_plugin_refine_item_state (self, app_tmp, cancellable, &error_local)) {
+			if (g_cancellable_is_cancelled (cancellable)) {
+				g_propagate_error (error, g_steal_pointer (&error_local));
+				return NULL;
+			}
+			g_warning ("Failed to refine %s when getting the list of apps to "
+				   "remove: %s; not adding the app...",
+				   gs_app_get_unique_id (app_tmp),
+				   error_local->message);
+			continue;
+		}
```

Comment:
It feels like this chunk of code could be refactored into a separate private function, and reuse it from `_get_list_for_install()`

Reply:
I get your point but without the warning (it's slightly different in each version) then we'd not be saving a lot (and I am not a fan of printing debug/warning messages from shared helper functions).

Reply:
Ok. No big deal anyway